C++/4. Advanced-Operators.md

```diff
@@ -2,11 +2,11 @@
 
 ## Start
 
-In this file I am going to teach you about advanced operators. There are three more types we are going to learn about - Assignment Operators, Relational Operators, and Logical Operators.
+In this file I am going to teach you about more advanced operators. There are three more types we are going to learn about - Compound-assignment operators, relational operators, and logical operators.
 
-## Assignment Operators
+## Coumpund-assignment Operators
 
-The Assignment Operators are basically shortcuts for making a variable be equal to a number (operator) itself:
+The Coumpund-assignment operators are basically shortcuts for assigning a variable with the result of applying an operator to itself and another value:
```

Comment:
I feel like my wording is awful here; suggestions?

Reply:
The **compound** assignment operators are shortcuts for assiigning a variable with the result of adding, multiplying, subtracting, dividing, or finding the remainder of that variable and another number

Reply:
I see my misspelling, but it's also more generic than that, since it applies to a lot of other operators (and C++ has quite a few). For example, `&=`:
```c++
auto are_all_above_5 = true;

for ( auto i = 0u; i < 10; ++i ) {
    are_all_above_5 &= i > 5;
}
```

Reply:
Bruh I have no idea what that operator means...

Reply:
"bitwise and"? `are_all_above_5 = are_all_above_5 & (i > 5)`

If I may ask, just to help me understand where you're coming from, how much C++ do you know and/or how long have you been using the language?

Reply:
> If I may ask, just to help me understand where you're coming from, how much C++ do you know and/or how long have you been using the language?

I have been using the language for like a year rn (maybe more). I know the basic knowledge needed to code like medium level projects. I was about to say I don't know things like the "bitwise and" which you don't need to learn but then I understood what you meant. You are talking about the "&", "|", and "!" right? I know those, but I thought you can only use them in loops and if statements...

P.S. What is the result to the above code you shared? `are_all_above_5 = are_all_above_5 & (i > 5)` What will the variable be equal to if you print it out rn?

Reply:
> ...but I thought you can only use them in loops and if statements...

Yeah, no, any operator can be used almost anywhere, and they merely yield a value. It doesn't matter how you got a value into an if statement, all that matters is what the value is.

> What is the result to the above code you shared?

I hoped that the variable's name would have been self-explaining, but ask yourself, are all values between [0, 9) greater than 5? You may notice that the range exceeds 5, so they are not all above 5.

Btw, I had left a comment under the loop that already answered your question.